patches/tModLoader/Terraria/Social/Steam/WorkshopHelper.TML.cs

```diff
@@ -366,168 +282,225 @@ private void ReleaseWorkshopQuery()
 				SteamedWraps.ReleaseWorkshopHandle(_primaryUGCHandle);
 			}
 
+			/////// Queries ////////////////////
+
 			/// <summary>
-			/// For direct information gathering of a particular mod/workshop item
+			/// For direct information gathering of particular mod/workshop items. Synchronous
 			/// </summary>
-			internal SteamUGCDetails_t FastQueryItem(ulong publishedId, bool queryChildren = false)
+			internal ModDownloadItem[] FastQueryItems()
 			{
-				TryRunQuery(SteamedWraps.GenerateSingleItemQuery(publishedId));
+				var numPages = Math.Ceiling(queryParameters.searchModIds.Length / (float)Constants.kNumUGCResultsPerPage);
+				var items = new ModDownloadItem[queryParameters.searchModIds.Length];
 
-				var pDetails = SteamedWraps.FetchItemDetails(_primaryUGCHandle, 0);
+				for (int i = 0; i < numPages; i++) {
+					var pageIds = queryParameters.searchModIds.Take(new Range(i * Constants.kNumUGCResultsPerPage, Constants.kNumUGCResultsPerPage * (i + 1) - 1));
 
-				// Some weird mod will have a super big m_unNumChildren and will cause game crash.
-				if (pDetails.m_unNumChildren > 1000) {
-					throw new OverflowException("Numbers of dependencies exceeds 1000. Dependencies from Steam Workshop: " + pDetails.m_unNumChildren);
-				}
+					if (!TryRunQuery(SteamedWraps.GenerateDirectItemsQuery(pageIds.Select(x => x.m_ModPubId).ToArray()))) {
+						Logging.tML.Error($"Unexpectedly failed to query information reqarding items {pageIds}.");
+						return null;
+					}
 
-				if (queryChildren) {
-					ugcChildren = SteamedWraps.FetchItemDependencies(_primaryUGCHandle, 0, pDetails.m_unNumChildren).Select(x => x.m_PublishedFileId).ToList();
-				}
+					for (int j = 0; j < i * Constants.kNumUGCResultsPerPage + _queryReturnCount; j++) {
+						items[j] = GenerateModDownloadItemFromQuery((uint)j);
+						items[j].UpdateInstallState();
```

Comment:
This is gonna crash if the above method returns null. As mentioned elsewhere, I don't think there's a reason why you would want `GenerateModDownloadItemFromQuery` to have a stale `InstallState` anyway